File: miniJobs/Domain/Dtos/JobDTO.cs
using Domain.Enums;

namespace Domain.Dtos;

public class JobDTO
{
    public int Id { get; set; }

    public string Name { get; set; }

    public string Description { get; set; }

    public int? ApplicationsDuration { get; set; }

    public JobStatus Status { get; set; }
    public int? RequiredEmployees { get; set; }

    public string CityName { get; set; }

    public string JobTypeName { get; set; }

    public DateTime Created { get; set; }

    public int CreatedBy { get; set; }
    public string EmployerFullName { get; set; }
    public bool DeletedByAdmin { get; set; }
}

File: miniJobs/UI/minijobs_admin/lib/models/job/job_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:minijobs_admin/enumerations/job_statuses.dart';

part 'job_dto.g.dart';

@JsonSerializable()
class JobDTO {
  int id;
  String name;
  String description;
  int? applicationsDuration;
  JobStatus? status;
  String cityName;
  String? jobTypeName;
  DateTime created;
  int createdBy;
  String employerFullName;
  bool deletedByAdmin;
  int? requiredEmployees;


  JobDTO(this.id,
      this.name,
      this.description,
      this.applicationsDuration,
      this.status,
      this.cityName,
      this.created,
      this.jobTypeName,
      this.createdBy,
      this.employerFullName,
      this.deletedByAdmin,
      this.requiredEmployees);

  factory JobDTO.fromJson(Map<String, dynamic> json) => _$JobDTOFromJson(json);

}